src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -839,9 +839,29 @@ void ApplicationManagerImpl::ConnectToDevice(const std::string& device_mac) {
 
 void ApplicationManagerImpl::OnHMIStartedCooperation() {
   LOG4CXX_AUTO_TRACE(logger_);
-  hmi_cooperating_ = true;
   MessageHelper::SendGetSystemInfoRequest(*this);
 
```

Comment:
@ydementieiev Check that your PR is actual as this function has more code in the current develop branch

Reply:
@AByzhynar  yes PR is actual, some of code move [here](https://github.com/smartdevicelink/sdl_core/pull/3269/commits/2d5949fecc86315df964b15a2244ef309a0f87e4#diff-6289bfe1988a528c2f109887d563c017R859)

Reply:
@AByzhynar @ydementieiev and also please note that this PR is based on the feature/persisting_hmi_capabilities branch, not on the current develop branch

Reply:
@LitvinenkoIra Rebase the feature/persisting_hmi_capabilities branch to the latest develop, otherwise you will have lots of logical issues at the end.

Reply:
@AByzhynar yeah this is planned activity right after merge of this PR